Makefile

```diff
@@ -125,11 +190,19 @@ fmt:
 vet:
 	go vet ./pkg/... ./cmd/...
 
+# Run go mod tidy against code
+tidy:
+	go mod tidy
+
+# Run go mod vendor against code
+vendor:
+	go mod vendor
+
 # Generate code
 generate: controller-gen
 	$(CONTROLLER_GEN) object:headerFile="./hack/boilerplate.go.txt" paths="./pkg/apis/..."
 
-build-controller-image: check_container_runtmime
+build-controller-image: check_container_runtmime fmt tidy vendor vet
```

Comment:
I wouldn't expect `tidy` and `vendor` to be called in this context as the dependencies shouldn't change that much.. what's the reason for adding them here?
and about `fmt` and `vet` - there's no reason to execute them per-image, right? how about adding them to build-all-images or push-all-images?

Reply:
> I wouldn't expect `tidy` and `vendor` to be called in this context as the dependencies shouldn't change that much.. what's the reason for adding them here? 

Yeah, I agree, I added them without thinking much about it. Indeed, I think the missing `bazel-generate` target was the actual reason for `tidy` and `vendor` IIRC.

> and about `fmt` and `vet` - there's no reason to execute them per-image, right? how about adding them to build-all-images or push-all-images?

I also thought to add them to the `build-all-images` but none of us use that target that much I would say. The usual workflow is more like: push all images the first time and then push the controller image (or any other image we are working on), then remove the controller pod and test the changes.

The main reason for adding them per image build was to make sure we do not submit code without correct formatting. Currently if you run `go fmt` in the project some of the files will be modified and this is not the first (nor the last) time it occurs. The same happened in the past with the vendor dirs or the bazel build files which were not regenerated/updated when needed. So this was an attempt to make sure everything is up to date before posting the PRs.

Reply:
Added the `generate-verify` target and it looks like it's working, the build failed 🤣